plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,235 @@
-        in
-        {
-          enable = mkEnableOption "incremental selection based on the named nodes from the grammar";
-          keymaps = {
-            initSelection = keymap "gnn";
-            nodeIncremental = keymap "grn";
-            scopeIncremental = keymap "grc";
-            nodeDecremental = keymap "grm";
-          };
-        };
+  settingsExample = {
+    additional_vim_regex_highlighting = true;
+    ensure_installed = true;
+    ignore_install = [ "rust" ];
+    indent = true;
+    parser_install_dir = "$XDG_DATA_HOME/nvim/treesitter";
+
+    highlight = {
+      enable = true;
+      disable = [ "rust" ];
+      custom_captures = { };
+    };
 
-      indent = mkEnableOption "tree-sitter based indentation";
-
-      folding = mkEnableOption "tree-sitter based folding";
-
-      languageRegister = mkOption {
-        type = with types; attrsOf (either str (listOf str));
-        description = ''
-          This is a wrapping of the `vim.treesitter.language.register` function.
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    incremental_selection = {
+      enable = true;
 
-      grammarPackages = mkOption {
-        type = with types; listOf package;
-        default = cfg.package.passthru.allGrammars;
-        description = "Grammar packages to install";
+      keymaps = {
+        init_selection = "gnn";
+        node_incremental = "grn";
+        scope_incremental = "grc";
+        node_decremental = "grm";
       };
+    };
+  };
 
-      moduleConfig = mkOption {
-        type = types.attrsOf types.anything;
-        default = { };
-        description = "This is the configuration for extra modules. It should not be used directly";
-      };
+  extraOptions = {
+    folding = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Whether to enable tree-sitter based folding.";
+    };
 
-      nixvimInjections = mkEnableOption "nixvim specific injections, like lua highlighting in extraConfigLua";
+    gccPackage = helpers.mkPackageOption {
+      name = "gcc";
+      default = pkgs.gcc;
+      description = ''
+        Which package (if any) to be added as the GCC compiler.
+        This is required to build grammars if you are not using `nixGrammars`.
+        To disable the installation of GCC, set this option to `null`.
+      '';
     };
-  };
 
-  config =
-    let
-      tsOptions = {
-        highlight = {
-          inherit (cfg) enable;
-          disable = if (cfg.disabledLanguages != [ ]) then cfg.disabledLanguages else null;
+    grammarPackages = mkOption {
+      type = with types; listOf package;
+      default = pkgs.vimPlugins.nvim-treesitter.passthru.allGrammars;
```

Comment:
I wonder if it's better to default to `cfg.package.passthru.allGrammars`? That'd probably work better if users (e.g.) wanted a very specific version of treesitter.

Should also consider if we need an assertion or something for when these options are used alongside a null `cfg.package`.

`package` is nullable to allow disabling installing the package, sometimes users do that in order to use the package from the environment.